content/blog_posts/articles/hmr-in-adonisjs.md

```diff
@@ -0,0 +1,106 @@
+---
+summary: Hot Module Replacement (HMR) is now available in AdonisJS.
+---
+
+We're thrilled to announce that HMR is now available in AdonisJS!
+
+In short: HMR now allows us to modify the code of our controllers and their dependencies without having to restart the entire dev server each time. This results in a much faster feedback loop and a significantly improved development experience.
+
+## Hot Hook
+
+We've developed [Hot-Hook](https://github.com/Julien-R44/hot-hook), a framework-agnostic library that enables HMR in Node.JS with ESM. The library works with any framework. It's also important to note that Hot-hook **does not perform any static analysis, heavy code transformations, AST parsing, or bundling**.
+
+The approach is very simple but works perfectly. 
+
+:::note
+Make sure to read the documentation of [Hot-Hook](https://github.com/Julien-R44/hot-hook) which contains a lot of information about how it works, configuration, and some key concepts of the library.
+:::
+
+### How does it work?
+
+We won't go deeply in the details of the implementation, but here's a summary of how it works: 
```

Comment:
```suggestion
Since ESM uses `URL`, we can leverage the **cache busting** mechanism to ensure fresh modules are loaded. Here's a summary of how it works: 
```